test/integration/scheduler/plugins/plugins_test.go

```diff
@@ -2561,7 +2442,34 @@ func initTestSchedulerForFrameworkTest(t *testing.T, testCtx *testutils.TestCont
 			t.Fatal(err)
 		}
 	}
-	return testCtx
+
+	// teardown will delete created resources and scheduler here, resources should be those
+	// that will affect the scheduling result, like nodes, pods, etc.. Namespaces should not be
+	// deleted here because it's created together with the apiserver, they should be deleted
+	// simultaneously or we'll have no namespace.
+	// This should only be called when you want to kill the scheduler alone, away from apiserver.
+	// For example, in scheduler integration tests, recreating apiserver is performance consuming,
+	// then shutdown the scheduler and recreate it between each test case is a better approach.
```

Comment:
This seems to be the explanation how the `ShutdownFunc` that gets returned by `initTestSchedulerForFrameworkTest` should be used.

That explanation should be in the function documentation of `initTestSchedulerForFrameworkTest`, not in the middle of its implementation.

Reply:
Moved to the head.